Core/Matcher/UserGroupMatcher.php

```diff
@@ -75,4 +88,25 @@ protected function findUserGroupsById(array $userGroupIds)
 
         return $userGroups;
     }
+
+    /**
+     * @param string[] $remoteContentIds
+     * @return UserGroup[]
+     */
+    protected function findUserGroupsByContentRemoteIds(array $remoteContentIds)
+    {
+        $userGroups = [];
+
+        foreach ($remoteContentIds as $remoteContentId) {
+            // return unique contents
+
+            // user service does not provide a method to load user groups via remote_id, but as user groups are content...
```

Comment:
See this comment. As the user service does not provide a method to load user group by remote_id, but a user group is stored as a content which supports remote id of course, I added this "hack" to first fetch the content by remote_id and then fetch the user group by the content's id. Again, slight performance impact for that use case, but at least we cover the use case :)